miden-lib/asm/miden/tx.masm

```diff
@@ -195,7 +195,55 @@ export.execute_foreign_procedure
     # reset the current account data offset to the native offset (2048)
     exec.kernel_proc_offsets::end_foreign_context_offset
     # => [offset, <values returned from the foreign procedure>]
-    
+
     syscall.exec_kernel_proc
     # => [<values returned from the foreign procedure>]
 end
+
+#! Updates the transaction expiration block delta.
+#!
+#! The input block_height_delta is added to the block reference number in order to determine an upper
+#! limit at which the transaction will be considered valid (not expired).
+#! This value can be later decreased, but not increased.
+#!
+#! Inputs: [block_height_delta, ...]
+#! Output: [...]
+#!
+#! Where:
+#! - block_height_delta is the desired expiration time delta (1 to 0xFFFF).
+export.update_expiration_block_delta
+    exec.kernel_proc_offsets::update_expiration_block_num_offset
+    # => [offset, expiration_delta, ...]
+
+    # pad the stack
+    push.0 movdn.2 push.0 movdn.2 padw swapw padw padw swapdw
+    # => [offset, expiration_delta, PAD(14)]
+
+    syscall.exec_kernel_proc
+
+    # clear the stack
+    dropw dropw dropw dropw
+end
+
+#! Gets the transaction expiration delta.
```

Comment:
I would update this to: "Returns the transaction expiration delta, or 0 if the delta has not been set."

But also, I just realized that we would return some really large value if the delta has not been set. This needs to be fixed in `ams/kernel/lib/tx.masm`.